pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -178,6 +185,20 @@ type ServiceMonitorSpec struct {
 	BodySizeLimit *ByteSize `json:"bodySizeLimit,omitempty"`
 }
 
+// ServiceMonitorStatus is the most recent observed status of the ServiceMonitor. Read-only.
+// More info:
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +k8s:openapi-gen=true
+type ServiceMonitorStatus struct {
+	// The list of workload resources(Prometheus or PrometheusAgent) the service monitor is link to.
+	// +patchMergeKey=resource
+	// +patchMergeKey=namespace
+	// +patchMergeKey=name
+	// +patchMergeStrategy=merge
```

Comment:
kubebuilder doesn't support patchMerge* annotations.
```suggestion
	// +listMapKey=group
	// +listMapKey=name
	// +listMapKey=namespace
	// +listMapKey=resource
	// +listType=map
	// +mapType=atomic
```